src/samplers/sampler.jl

```diff
@@ -32,7 +33,7 @@ assume(spl::Void, dist::Distribution, vn::VarName, vi::VarInfo) = begin
   if haskey(vi, vn)
     r = vi[vn]
   else
-    r = rand(dist)
+    r = init(dist)
```

Comment:
perhaps we should only do this for Hamiltonian samplers, since sometimes we need to sample from the prior and compute importance weights. 

```julia
isa(eltypes(spl), Hamiltonian) ? init(dist) : rand(dist)
```

```